source/grid/grid_generator.cc

```diff
@@ -1910,6 +1910,353 @@ namespace GridGenerator
     tria.create_triangulation(points, cells, SubCellData());
   }
 
+
+
+  template <>
+  void plate_with_a_hole(Triangulation<1> & /*tria*/,
+                         const double /*inner_radius*/,
+                         const double /*outer_radius*/,
+                         const double /*pad_bottom*/,
+                         const double /*pad_top*/,
+                         const double /*pad_left*/,
+                         const double /*pad_right*/,
+                         const Point<1> /*center*/,
+                         const types::manifold_id /*polar_manifold_id*/,
+                         const types::manifold_id /*tfi_manifold_id*/,
+                         const double /*L*/,
+                         const unsigned int /*n_slices*/,
+                         const bool /*colorize*/)
+  {
+    Assert(false, ExcNotImplemented());
+  }
+
+
+  namespace internal
+  {
+    // helper function to check if point is in 2D box
+    bool inline point_in_box(const Point<2> &p,
+                             const Point<2> &c,
+                             const double    radius)
+    {
+      return (std::abs(p[0] - c[0]) < radius) &&
+             (std::abs(p[1] - c[1]) < radius);
+    }
+
+    bool inline point_in_box(const Point<3> &p,
+                             const Point<3> &center,
+                             const double    radius)
+    {
+      return point_in_box(Point<2>(p[0], p[1]),
+                          Point<2>(center[0], center[1]),
+                          radius);
+    }
+
+    /**
+     * Keep the main 2D implementation internal as we will use
+     * material id in 3D case, whereas the actual 2D function
+     * will reset material id in addition to this one.
+     */
+    void plate_with_a_hole(Triangulation<2> &       tria,
+                           const double             inner_radius,
+                           const double             outer_radius,
+                           const double             pad_bottom,
+                           const double             pad_top,
+                           const double             pad_left,
+                           const double             pad_right,
+                           const Point<2>           new_center,
+                           const types::manifold_id polar_manifold_id,
+                           const types::manifold_id tfi_manifold_id,
+                           const double             L,
+                           const unsigned int /*n_slices*/,
+                           const bool colorize)
+    {
+      Assert((pad_bottom > 0 || pad_top > 0 || pad_left > 0 || pad_right > 0),
+             ExcMessage("At least one padding parameter has to be non-zero."));
```

Comment:
Would it be wrong to just return the result of `GridGenerator::hyper_cube_with_cylindrical_hole` in this scenario?

Reply:
I don't think it would be wrong, but I did not test this.